### ISSUE TITLE
Can Violist analyze android-6.0 APKs?

### ISSUE THREAD

**Issue by moonZHH** [marapapman/Violist#2]
@marapapman 
Hi!
I wonder that can Violist analyze android-6.0 APKs?

**Comment by marapapman** [comment]
We did not try it, so we do not know.

**Comment by moonZHH** [comment]
@marapapman 
Thanks for your reply! Maybe I need to try it myself.

**Comment by moonZHH** [comment]
@marapapman 
Hi!
I have another problem. Here is my code snapshot in a test apk file.
protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        String version = "java";
        String seed = "_test";
        String result = version;
        Logger.reportString(result,"1");
        for(int i = 0; i < 5; i++) {
            result = result + seed;
            Logger.reportString(result,"2");
        }
    }
I just wonder why Violist can not get the result form the statement Logger.reportString(result,"2"); ?
But when I move this code snapshot into a test java file, Violist works this time.

**Comment by winsonlyu** [comment]
Hi,
Is there any results from Logger.reportString(result,"1") for the test apk
file?

Yingjun

**Comment by moonZHH** [comment]
@winsonlyu 
Hi!
Actually, yes.
Violist can get result from statement Logger.reportString(result,"1"), but statement  Logger.reportString(result,"2") has no result.

**Comment by moonZHH** [comment]
Actually, if I annotate the following code snapshot, the previous problem can be solved. Will it cause any other problems by this annotation?
//Android Crash 
/*
if(rd.getAllDef().get(n.getOffset().toString()).startsWith("$"))
{
    vlist.add(v);
    continue;
}
*/

**Comment by winsonlyu** [comment]
Yes, it might cause some runtime exception for some apps. But in the meanwhile, you can get more precise results from Violist. 
In our implementation, I try to make our tool as safe as possible. So I choose not to handle those cases. But you can adjust it. As long as it does not raise any exception, the analysis result will be correct.

Yingjun
